### PR TITLE
Add an updated bottleneck tut from Konrad Lohse

### PR DESCRIPTION
**Pull request by hyanwong**
The old one had lots of missing stuff

### CODE REVIEW

**Comment by hyanwong** [comment]
Failing because of timing out - requires an updated tskit version to be released with https://github.com/tskit-dev/tskit/pull/1496 merged

File: bottlenecks.md
Comment (benjeffery) [inline bottlenecks.md:21]:
```suggestion
Here we simulate a single sample of n=12 from a population that underwent an instantaneous bottleneck 0.4 * 2N generations ago. Note that since msprime is assuming a diploid population we set the initial population size to 1/2 to obtain coalescence times scaled in units of $2N_e$ generations:
```

File: bottlenecks.md
Comment (benjeffery) [inline bottlenecks.md:23]: This will make things run on time for now:
```suggestion
  ```{code-cell} ipython3
  :"tags": ["remove_input"]
  # REMOVE WHEN TSKIT 0.3.7 RELEASED
  import tskit
  import functools
  tskit.metadata.parse_metadata_schema = functools.lru_cache(maxsize=128,(tskit.metadata.parse_metadata_schema)
    ```
```

File: bottlenecks.md
Comment (benjeffery) [inline bottlenecks.md:57]:
```suggestion
Bunnefeld et al (2016) derive the total length of n-ton branches under an instantaneous bottleneck using a recursion for the generating function of genealogies. Assuming a sample size of $n=4$, $B=4$ and $T=0.4$, the SFS is:
```

File: bottlenecks.md
Comment (benjeffery) [inline bottlenecks.md:192]:
```suggestion
Bottlenecks can have a substantial effect on the site frequency spectrum (SFS). The SFS is a fundamental summary of sequence variation that forms the basis of many modern inference approaches (e.g. sweepfinder, DFE-alpha, dadi). in the absence of linkage information the SFS is a lossless summary, i.e. any summary of sequence variation that ignores linkage (e.g. pairwise measures of diversity and divergence, $F_{st}$, Tajima's D etc) are summaries of the SFS (Achaz 2009). The SFS is convenient analytically, since it depends only on the mean length and frequency of genealogical branches. For many demographic models of interest the mean length of n-ton branches can be derived either using coalescent theory (Chen 2011) or diffusion equations (Gutenkunst 2009). A number of composite likelihood approaches have been developed based on either analytic results for the SFS or approximate obtained from coalescent simulations (Gutenkunst 2009, Excoffier 2002). We can use msprime to obtain the SFS for a sample of $n=20$ for a range of bottleneck strengths:
```

**Review by benjeffery — APPROVED** [review]
Added a suggestion here to get this merged before the needed tskit release.

File: bottlenecks.md
Comment (hyanwong) [reply bottlenecks.md:192]: We should probably replace "site frequency spectrum" by "allele frequency spectrum" anyway, since that's what we use in the tskit docs

**Comment by hyanwong** [comment]
> Added a suggestion here to get this merged before the needed tskit release.

Thanks Ben. Happy to wait until then, or stick that in and merge now - what do you think's best?

**Comment by benjeffery** [comment]
Best to get this merged now.

**Review by jeromekelleher — APPROVED** [review]
LGTM

**Comment by hyanwong** [comment]
Fixed the lru_cache hack, so I guess this can be merged now?

**Pull request merged**
